crates/forge-runner/src/test_execution_syscall_handler.rs

```diff
@@ -785,6 +785,16 @@ fn write_call_contract_response(
                 .read_only_segments
                 .allocate(vm, &ret_data.iter().map(Into::into).collect())?;
 
+            cheatable_syscall_handler
+                .cheatnet_state
+                .used_resources
+                .vm_resources += &call_output.used_resources.vm_resources;
+            cheatable_syscall_handler
+                .cheatnet_state
+                .used_resources
+                .syscall_counter
+                .extend(call_output.used_resources.syscall_counter);
```

Comment:
We do have an extending function in an other place, why is this happening here as well?

Reply:
I guess this one is for the VMs created by blockifier when calling a contract, the other one is for the top level VM which captures only the test code gas

Reply:
that makes sense indeed, @war-in is that the case? could we somehow indicate that with the code?

Reply:
Yup, that's the case. I thought it should be obvious since the `cheatnet_state` keeps `used_resources` and adds resources used by the call.
But I will describe `CheatnetState.used_resources` to be explicit